src/main/java/com/nutomic/syncthingandroid/service/SyncthingRunnable.java

```diff
@@ -399,4 +379,56 @@ private void trimLogFile() {
         }
     }
 
+    private HashMap<String, String> buildEnvironment(SharedPreferences sp) {
+        HashMap<String, String> targetEnv = new HashMap<String, String>();
+        // Set home directory to data folder for web GUI folder picker.
+        targetEnv.put("HOME", Environment.getExternalStorageDirectory().getAbsolutePath());
+        targetEnv.put("STTRACE", sp.getString("sttrace", ""));
+        File externalFilesDir = mContext.getExternalFilesDir(null);
+        if (externalFilesDir != null)
+            targetEnv.put("STGUIASSETS", externalFilesDir.getAbsolutePath() + "/gui");
+        targetEnv.put("STNORESTART", "1");
+        targetEnv.put("STNOUPGRADE", "1");
+        // Disable hash benchmark for faster startup.
+        // https://github.com/syncthing/syncthing/issues/4348
+        targetEnv.put("STHASHING", "minio");
+        if (sp.getBoolean("use_tor", false)) {
+            targetEnv.put("all_proxy", "socks5://localhost:9050");
+            targetEnv.put("ALL_PROXY_NO_FALLBACK", "1");
+        }
+        if (sp.getBoolean("use_legacy_hashing", false))
+            targetEnv.put("STHASHING", "standard");
+        putCustomEnvironmentVariables(targetEnv, sp);
+        return targetEnv;
+    }
+
+    private Process setupAndLaunch(HashMap<String, String> env) throws IOException {
+        Process process = null;
+
+        ProcessBuilder pb = (mUseRoot)
+                ? new ProcessBuilder("su")
+                : new ProcessBuilder(mCommand);
```

Comment:
Can you move this into the `if (mUseRoot)` statement? I don't think there's a reason to have it seperately.

Reply:
No, because the ProcessBuilder has to be started in the else branch as well.

Reply:
Yes, but you can define seperate ProcessBuilders in the if/else branches. Much simpler than having 2 conditionals that check the exact same condition.

Reply:
aaah. ok. :) Got it.